react/hooks/hydration.tsx

```diff
@@ -0,0 +1,24 @@
+import { useRef } from 'react'
+
+/** Gets HTML content from elements rendered on the server, before being
+ * hydrated on the client (and whether they have been SSRendered at all). */
+export const useDehydratedContent = (hydrationId: string) => {
+  /** On both cases below, the use of useRef is to guarantee that only the values
+   * that it has been initialized with is being considered, to guarantee that it's
+   * only taking into account the HTML that's coming from the server, instead of
+   * eventual elements that might be included later.
+   */
+  const dehydratedElement = useRef(
+    window?.document?.querySelector?.(`[data-hydration-id="${hydrationId}"]`)
```

Comment:
are you sure this is executed only on the first render? have you tested?

Reply:
the querySelector is still called on every render.
If you want to avoid that, you might want to use useState with lazy init

`const [dehydratedElement] = useState(
    () => window?.document?.querySelector?.(`[data-hydration-id="${hydrationId}"]`)
)`

Reply:
Or do like this:
```
const dehydratedElement = useRef(null)
if (dehydratedElement.current == null) {
  dehydratedElement.current = window?.document?.querySelector?.(`[data-hydration-id="${hydrationId}"]`)
}
```

Reply:
gonna go with state